src/EventSourcing/Config/EventSourcingModule.php

```diff
@@ -114,6 +118,14 @@ public static function create(AnnotationFinder $annotationRegistrationService, I
             $aggregateToStreamMapping[$aggregateWithCustomStream] = $attribute->getName();
         }
 
+        $aggregateTypeMapping = [];
+        foreach ($annotationRegistrationService->findAnnotatedClasses(AggregateType::class) as $aggregateWithCustomType) {
+            /** @var Stream $attribute */
+            $attribute = $annotationRegistrationService->getAttributeForClass($aggregateWithCustomType, AggregateType::class);
+
+            $aggregateTypeMapping[$aggregateWithCustomType] = $attribute->getName();
+        }
```

Comment:
Great that you've analyzed the code, to make use of `AnnotationFinder` :) 

Things that are done in `Module::create` method are cached, so it's best place to do heavy lifting stuff :)